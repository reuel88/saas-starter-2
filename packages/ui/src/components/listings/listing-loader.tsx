import { FC } from "react";
import { Card, CardContent, CardFooter, CardHeader } from "../card";
import { Skeleton } from "../ui/skeleton";

type ListingLoaderProps = {};

export const ListingLoader: FC<ListingLoaderProps> = () => {
  return (
    <Card>
      <CardHeader>
        {/* Title */}
        <Skeleton className="h-6" />

        {/* Price & Location */}
        <div className="flex h-7 items-center">
          <Skeleton className="h-5 w-full" />
        </div>
      </CardHeader>

      <CardContent className="space-y-6">
        {/* Image */}
        <Skeleton className="aspect-video" />

        {/* Description */}
        <div className="space-y-1.5 pt-1.5">
          <Skeleton className="h-3.5 w-full" />
          <Skeleton className="h-3.5 w-full" />
          <Skeleton className="h-3.5 w-3/4" />
        </div>
      </CardContent>

      {/* CTA */}
      <CardFooter className="flex gap-4">
        <Skeleton className="h-10 w-20" />
        <Skeleton className="h-10 w-20" />
      </CardFooter>
    </Card>
  );
};
