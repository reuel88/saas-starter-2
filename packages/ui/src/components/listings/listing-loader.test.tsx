import { render } from "@testing-library/react";
import { ListingLoader } from "./listing-loader";

describe("ListingLoader", () => {
  it("should render correctly", () => {
    // Arrange

    // Act
    const { baseElement } = render(<ListingLoader />);

    // Assert
    expect(baseElement).toMatchSnapshot();
  });
});
