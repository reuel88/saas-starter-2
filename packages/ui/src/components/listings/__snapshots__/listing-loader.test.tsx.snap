// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ListingLoader > should render correctly 1`] = `
<body>
  <div>
    <div
      class="rounded-lg border bg-card text-card-foreground shadow-sm"
    >
      <div
        class="flex flex-col space-y-1.5 p-6"
      >
        <div
          class="animate-pulse rounded-md bg-muted h-6"
        />
        <div
          class="flex h-7 items-center"
        >
          <div
            class="animate-pulse rounded-md bg-muted h-5 w-full"
          />
        </div>
      </div>
      <div
        class="p-6 pt-0 space-y-6"
      >
        <div
          class="animate-pulse rounded-md bg-muted aspect-video"
        />
        <div
          class="space-y-1.5 pt-1.5"
        >
          <div
            class="animate-pulse rounded-md bg-muted h-3.5 w-full"
          />
          <div
            class="animate-pulse rounded-md bg-muted h-3.5 w-full"
          />
          <div
            class="animate-pulse rounded-md bg-muted h-3.5 w-3/4"
          />
        </div>
      </div>
      <div
        class="items-center p-6 pt-0 flex gap-4"
      >
        <div
          class="animate-pulse rounded-md bg-muted h-10 w-20"
        />
        <div
          class="animate-pulse rounded-md bg-muted h-10 w-20"
        />
      </div>
    </div>
  </div>
</body>
`;
